src/Compilers/Core/Portable/SourceGeneration/Nodes/BatchNode.cs

```diff
@@ -50,22 +53,7 @@ public NodeStateTable<ImmutableArray<TInput>> UpdateStateTable(DriverStateTable.
 
             var stopwatch = SharedStopwatch.StartNew();
 
-            var sourceValuesBuilder = ArrayBuilder<TInput>.GetInstance();
-            var sourceInputsBuilder = newTable.TrackIncrementalSteps ? ArrayBuilder<(IncrementalGeneratorRunStep InputStep, int OutputIndex)>.GetInstance() : null;
-
-            foreach (var entry in sourceTable)
-            {
-                // At this point, we can remove any 'Removed' items and ensure they're not in our list of states.
-                if (entry.State != EntryState.Removed)
-                    sourceValuesBuilder.Add(entry.Item);
-
-                // However, regardless of if the entry was removed or not, we still keep track of its step information
-                // so we can accurately report how long it took and what actually happened (for testing validation).
-                sourceInputsBuilder?.Add((entry.Step!, entry.OutputIndex));
-            }
-
-            var sourceValues = sourceValuesBuilder.ToImmutableAndFree();
-            var sourceInputs = newTable.TrackIncrementalSteps ? sourceInputsBuilder!.ToImmutableAndFree() : default;
+            var (sourceValues, sourceInputs) = GetValuesAndInputs(sourceTable, previousTable, newTable);
 
             if (previousTable.IsEmpty)
```

Comment:
the logic of htis large if/else is entirely unchanged.  git-diff is just doing a bad job.

Reply:
When I turn off whitespace, I see no changes here.

Reply:
![image](https://user-images.githubusercontent.com/4564579/176061585-bffe7821-9ff4-4472-b679-71f3ec39578f.png)

weird.  this is waht i see :)

Reply:
![image](https://user-images.githubusercontent.com/2371880/176062476-fa2b32e2-d182-4645-b5db-ab4cd8110f22.png)
I see you commenting on a line with no changes.

Reply:
wacky.  i have no idea what's up with github then.